src/test/java/uk/gov/ons/census/casesvc/service/FulfilmentRequestProcessorTest.java

```diff
@@ -0,0 +1,110 @@
+import static uk.gov.ons.census.casesvc.testutil.DataUtils.getRandomCase;
+import static uk.gov.ons.census.casesvc.testutil.DataUtils.getTestResponseManagementEvent;
+
+import java.util.Optional;
+import java.util.UUID;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.ons.census.casesvc.logging.EventLogger;
+import uk.gov.ons.census.casesvc.model.dto.EventDTO;
+import uk.gov.ons.census.casesvc.model.dto.FulfilmentRequestDTO;
+import uk.gov.ons.census.casesvc.model.dto.ResponseManagementEvent;
+import uk.gov.ons.census.casesvc.model.entity.Case;
+import uk.gov.ons.census.casesvc.model.repository.CaseRepository;
+
+@RunWith(MockitoJUnitRunner.class)
+public class FulfilmentRequestProcessorTest {
+
+  @Mock private CaseProcessor caseProcessor;
+
+  @Mock private CaseRepository caseRepository;
+
+  @Mock private EventLogger eventLogger;
+
+  @InjectMocks FulfilmentRequestProcessor underTest;
+
+  @Test
+  public void testGoodFulfilmentRequest() {
+    ResponseManagementEvent managementEvent = getTestResponseManagementEvent();
+    FulfilmentRequestDTO expectedFulfilmentRequest =
+        managementEvent.getPayload().getFulfilmentRequest();
+
+    // Given
+    Case expectedCase = getRandomCase();
+    expectedFulfilmentRequest.setCaseId(expectedCase.getCaseId().toString());
+
+    when(caseRepository.findByCaseId(UUID.fromString(expectedFulfilmentRequest.getCaseId())))
+        .thenReturn(Optional.of(expectedCase));
+
+    // when
+    underTest.processFulfilmentRequest(managementEvent);
+
+    // then
+    verify(eventLogger, times(1))
+        .logFulfilmentRequestedEvent(
+            expectedCase,
+            expectedCase.getCaseId(),
+            managementEvent.getEvent().getDateTime(),
+            "Fulfilment Request Received",
+            FULFILMENT_REQUESTED,
+            expectedFulfilmentRequest,
+            managementEvent.getEvent());
+  }
+
+  @Test(expected = RuntimeException.class)
+  public void testCaseIdNotFound() {
+    // GIVEN
+    ResponseManagementEvent managementEvent = getTestResponseManagementEvent();
+    managementEvent.getPayload().getFulfilmentRequest().setCaseId(UUID.randomUUID().toString());
+    UUID expectedCaseIdNotFound =
+        UUID.fromString(managementEvent.getPayload().getFulfilmentRequest().getCaseId());
+    String expectedErrorMessage = String.format("Case ID '%s' not found!", expectedCaseIdNotFound);
+
+    try {
+      // WHEN
+      underTest.processFulfilmentRequest(managementEvent);
+    } catch (RuntimeException re) {
+      // THEN
+      assertThat(re.getMessage()).isEqualTo(expectedErrorMessage);
+      throw re;
+    }
+  }
+
+  @Test(expected = RuntimeException.class)
+  public void testNullDateTime() {
+    ResponseManagementEvent managementEvent = getTestResponseManagementEvent();
```

Comment:
Still works fine, but might be worth using the tailored test object for fulfilment here?

```suggestion
    ResponseManagementEvent managementEvent = getTestResponseManagementFulfilmentRequestedEvent();
```

Reply:
Receipt and refusal tests both use the generic method call so just following the consistency!